homeassistant/components/trend/binary_sensor.py

```diff
@@ -52,23 +52,39 @@
     CONF_INVERT,
     CONF_MAX_SAMPLES,
     CONF_MIN_GRADIENT,
+    CONF_MIN_SAMPLES,
     CONF_SAMPLE_DURATION,
     DOMAIN,
 )
 
 _LOGGER = logging.getLogger(__name__)
 
-SENSOR_SCHEMA = vol.Schema(
-    {
-        vol.Required(CONF_ENTITY_ID): cv.entity_id,
-        vol.Optional(CONF_ATTRIBUTE): cv.string,
-        vol.Optional(CONF_DEVICE_CLASS): DEVICE_CLASSES_SCHEMA,
-        vol.Optional(CONF_FRIENDLY_NAME): cv.string,
-        vol.Optional(CONF_INVERT, default=False): cv.boolean,
-        vol.Optional(CONF_MAX_SAMPLES, default=2): cv.positive_int,
-        vol.Optional(CONF_MIN_GRADIENT, default=0.0): vol.Coerce(float),
-        vol.Optional(CONF_SAMPLE_DURATION, default=0): cv.positive_int,
-    }
+
+def _validate_min_max(data: dict[str, Any]) -> dict[str, Any]:
+    if (
+        CONF_MIN_SAMPLES in data
+        and CONF_MAX_SAMPLES in data
+        and data[CONF_MAX_SAMPLES] < data[CONF_MIN_SAMPLES]
```

Comment:
The condition does not match the error message.
If the condition is correct, the error message should be:
"min_samples must be smaller than or equal to max_samples"

Reply:
D'oh, could have checked this. Fixed, thanks!